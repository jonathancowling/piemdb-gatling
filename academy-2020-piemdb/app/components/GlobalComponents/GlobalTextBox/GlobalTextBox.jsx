import React from 'react';
import TextField from '@material-ui/core/TextField';
import PropTypes from 'prop-types';

const GlobalTextBox = ({
  inputProps,
  id,
  label,
  rows,
  value,
  onChange,
  onReturnPressed,
}) => {
  const keyPress = (event) => {
    if (onReturnPressed && event.keyCode === 13) {
      onReturnPressed();
    }
  };
  return (
    <TextField
      inputProps={inputProps}
      id={id}
      label={label}
      multiline
      rows={rows}
      value={value}
      fullWidth
      variant={'outlined'}
      onChange={onChange}
      onKeyDown={keyPress}
    />
  );
};

GlobalTextBox.propTypes = {
  inputProps: PropTypes.object,
  id: PropTypes.string.isRequired,
  label: PropTypes.string,
  rows: PropTypes.number,
  value: PropTypes.string,
  onChange: PropTypes.func,
  onReturnPressed: PropTypes.func,
};

GlobalTextBox.defaultProps = {
  inputProps: {},
  rows: 1,
  onChange: () => { },
  onReturnPressed: null,
};

export default GlobalTextBox;
