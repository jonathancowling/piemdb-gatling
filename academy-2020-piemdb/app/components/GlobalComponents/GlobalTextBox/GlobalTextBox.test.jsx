import React from 'react';
import {
  render, screen, cleanup,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import GlobalTextBox from './GlobalTextBox.jsx';

afterEach(cleanup);

describe('/components/GlobalComponents/GlobalTextBox/GlobalTextBox.jsx', () => {
  it('should return a text box with the correct default text', () => {
    render(<GlobalTextBox
      inputProps={{ 'data-testid': 'test-box' }}
      id="test-box-id"
      label="test-label"
      value="Some text here"
    />);

    const Box = screen.getByTestId('test-box');
    expect(Box).toHaveTextContent('Some text here');
  });

  it('should return a text box with the correct label', () => {
    render(<GlobalTextBox
      inputProps={{ 'data-testid': 'test-box' }}
      id="test-box-id"
      label="test-label"
      value="Some text here"
    />);

    const Box = screen.getByLabelText('test-label');
    expect(Box).toHaveTextContent('Some text here');
  });

  it('snapshot should match for default values', () => {
    const snapshot = renderer.create(<GlobalTextBox
      inputProps={{ 'data-testid': 'test-box' }}
      id="test-box-id"
    />).toJSON();

    expect(snapshot).toMatchSnapshot();
  });

  it('snapshot should match for different than default values', () => {
    const snapshot = renderer.create(<GlobalTextBox
      inputProps={{ 'data-testid': 'test-box' }}
      id="test-box-id"
      label="test-label"
      rows={4}
      value="Some text here"
    />).toJSON();

    expect(snapshot).toMatchSnapshot();
  });

  it('should complete the on change function when a change occurs', () => {
    const mockFunc = jest.fn();

    render(<GlobalTextBox
      inputProps={{ 'data-testid': 'test-box' }}
      id="test-box-id"
      onChange={mockFunc}
    />);

    const userInput = 'Hello, World!';
    userEvent.type(screen.getByTestId('test-box'), userInput);

    expect(mockFunc).toHaveBeenCalledTimes(userInput.length);
  });
});
