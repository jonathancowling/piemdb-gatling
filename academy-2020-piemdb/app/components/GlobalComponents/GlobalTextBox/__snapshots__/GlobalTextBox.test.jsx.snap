// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/components/GlobalComponents/GlobalTextBox/GlobalTextBox.jsx snapshot should match for default values 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  onKeyDown={[Function]}
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
    onClick={[Function]}
  >
    <textarea
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
      data-testid="test-box"
      disabled={false}
      id="test-box-id"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      rows={1}
    />
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="PrivateNotchedOutline-legend-10"
        style={
          Object {
            "width": 0.01,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`/components/GlobalComponents/GlobalTextBox/GlobalTextBox.jsx snapshot should match for different than default values 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  onKeyDown={[Function]}
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
    data-shrink={true}
    htmlFor="test-box-id"
    id="test-box-id-label"
  >
    test-label
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
    onClick={[Function]}
  >
    <textarea
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
      data-testid="test-box"
      disabled={false}
      id="test-box-id"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      rows={4}
      value="Some text here"
    />
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="PrivateNotchedOutline-legendLabelled-11 PrivateNotchedOutline-legendNotched-12"
      >
        <span>
          test-label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
